compiler/rustc_typeck/src/collect.rs

```diff
@@ -2404,6 +2424,16 @@ fn explicit_predicates_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredicat
             }
         }
     } else {
+        if matches!(def_kind, DefKind::AnonConst) && tcx.lazy_normalization() {
+            // Provide predicates of parent item of cg defaults manually as `generics_of`
```

Comment:
Again, can you give an example where there are some where clauses and show what is going on here in terms of the Rust code?